### PR TITLE
fix Issue 22960 - importC: K&R-style functions assume variadic callin…

### PR DESCRIPTION
**Pull request by WalterBright**
…g convention

I.e. the pre function prototype C style. While such functions are called the same way as the prototyped version, the callee doesn't get the stack frame setup if it's the K&R style.

### CODE REVIEW

File: compiler/src/dmd/cparse.d [hunk 1]
```diff
@@ -2081,8 +2081,8 @@ final class CParser(AST) : Parser!AST
             auto pl = ft.parameterList;
             if (pl.varargs != AST.VarArg.none && pl.length)
                 error("function identifier-list cannot end with `...`");
-            ft.parameterList.varargs = AST.VarArg.variadic;     // but C11 allows extra arguments
-            importBuiltins = true;                              // will need __va_list_tag
+            ft.parameterList.varargs = AST.VarArg.KRvariadic;   // but C11 allows extra arguments
+            //importBuiltins = true;                            // will need __va_list_tag
```
Comment (RazvanN7) [inline compiler/src/dmd/cparse.d:2085]: Just delete this line.

File: compiler/src/dmd/cparse.d [hunk 2]
```diff
@@ -3046,8 +3046,8 @@ final class CParser(AST) : Parser!AST
         if (token.value == TOK.rightParenthesis)        // func()
         {
             nextToken();
-            importBuiltins = true;                              // will need __va_list_tag
-            return AST.ParameterList(parameters, AST.VarArg.variadic, varargsStc);
+            //importBuiltins = true;                    // will need __va_list_tag
```
Comment (RazvanN7) [inline compiler/src/dmd/cparse.d:3049]: ditto

File: compiler/src/dmd/cparse.d (hunk 1 above)
Comment (WalterBright) [reply compiler/src/dmd/cparse.d]: ok

File: compiler/src/dmd/cparse.d (hunk 2 above)
Comment (WalterBright) [reply compiler/src/dmd/cparse.d]: ok

**Comment by WalterBright** [comment]
ping @RazvanN7

File: compiler/src/dmd/frontend.h [hunk 3]
```diff
@@ -8873,7 +8874,9 @@ struct Id final
     static Identifier* naked;
     static Identifier* thread;
     static Identifier* vector_size;
+    static Identifier* noinline;
     static Identifier* noreturn;
+    static Identifier* _deprecated;
```
Comment (RazvanN7) [inline compiler/src/dmd/frontend.h:8879]: These header additions don't seem to be related to this PR.

Comment (WalterBright) [reply compiler/src/dmd/frontend.h:8879]: It's because I forgot to update frontend.h on a previous PR.

**Review by RazvanN7 — APPROVED** [review]
